From 3d6a441fd35e5891b0470af8c1bf216965116f31 Mon Sep 17 00:00:00 2001
From: Michael Niedermayer <michael@niedermayer.cc>
Date: Tue, 6 Oct 2026 07:31:35 +0200
Subject: [PATCH] avformat/rtpdec_rfc4175: Clear the frame buffer

The fix is the one recommended in the report.

Fixes: use of uninitialized memory
Fixes: V8a3e2cGQI7w
Fixes: AISLE-2026-0111-00477
Use of uninitialized memory Replicated through UnModified FFmpeg with Valgrind
Found-by: Joshua Rogers <joshua.rogers@aisle.com>
(cherry picked from commit 3d6a441fd35e5891b0470af8c1bf216965116f31)
---
 libavformat/rtpdec_rfc4175.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/rtpdec_rfc4175.c b/libavformat/rtpdec_rfc4175.c
index b49fc55d2dc..41bf4a2c5dc 100644
--- a/libavformat/rtpdec_rfc4175.c
+++ b/libavformat/rtpdec_rfc4175.c
@@ -260,7 +260,7 @@ static int rfc4175_handle_packet(AVFormatContext *ctx, PayloadContext *data,
         }
 
         if (!data->frame)
-            data->frame = av_malloc(data->frame_size);
+            data->frame = av_mallocz(data->frame_size + AV_INPUT_BUFFER_PADDING_SIZE);
 
         data->timestamp = *timestamp;
 
-- 
2.53.0

